src/storybook/story-test-runner/open-story/hermione-open-story.ts

```diff
@@ -0,0 +1,159 @@
+import type EventEmitter from "events";
+
+export type StoryLoadResult = {
+    rootSelector: string;
+    playFunctionError?: string;
+    loadError?: string;
+    notInjected?: true;
+};
+
+type HTMLElement = {
+    innerText: string;
+};
+
+type StorybookWindow = Window &
+    typeof globalThis & {
+        __HERMIONE_OPEN_STORY__: (storyId: string, remountOnly: boolean, done: (result: string) => void) => void;
+        __STORYBOOK_ADDONS_CHANNEL__: EventEmitter;
+        __STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__: Record<
+            string,
+            {
+                calls: Array<{
+                    exception?: {
+                        message: string;
+                    };
+                }>;
+            }
+        >;
+    };
+
+export async function inject(browser: WebdriverIO.Browser): Promise<void> {
+    await browser.execute(`window.__HERMIONE_OPEN_STORY__ = ${openStoryScript.toString()}`);
+}
+
+export async function execute(
```

Comment:
This includes reinject with retry (in case first test in browser session fails to inject script)